tests/test_bookmarks.py

```diff
@@ -0,0 +1,211 @@
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_messages = [record.get('data') for record in first_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in second_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+                if self.is_incremental(stream):
+
+                    self.assertEqual(first_bookmark_key_value, second_bookmark_key_value)
+
+                    # if the stream is child stream, then collect parent stream records for collecting ids
+                    if self.parent_child_mapping.get(stream):
+                        parent_records = [record.get('data') for record in first_sync_records.get(self.parent_child_mapping.get(stream) + 's').get('messages')
+                                        if record.get('action') == 'upsert']
+
+                    # NOTE: Sample bookmark:
+                    #   "ad_accounts": {
+                    #       "updated_at(parent_organization_id:organization-id-123)": "2020-01-01T00:00:00Z"
+                    #   }
+                    for key, value in first_bookmark_key_value.items():
+                        # partial split
+                        partial_splitted = key.split('parent_')
+
+                        # if the length of split is greater than 1, the bookmark is child stream
+                        if len(partial_splitted) > 1:
+                            splitted = partial_splitted[1].split(':')
+                            # parent value
+                            parent_value = splitted[0]
+                            # get parent record and remove last char ')' as we have splitted
+                            parent_record = splitted[1][:-1]
+
+                            # verify the parent in the bookmark is as expected
+                            self.assertEqual(parent_value, self.parent_child_mapping.get(stream) + '_id')
+
+                            # collect parent ids
+                            parent_ids = []
+                            for parent_rec in parent_records:
+                                parent_ids.append(parent_rec.get('id'))
+
+                            # verify the id in the bookmark is present in the parent ids
+                            self.assertTrue(parent_record in parent_ids, 'The bookmark contains other \'id\' rather than parent\'s \'id\'')
```

Comment:
assertTrue will not provide usable information. We'd want to see the id that failed. I prefer to avoid assertTrue/assertFalse at all costs and instead use other assertions. In this case you could use assertIn. Another acceptable option would be to log the list and record in the error msg.

Reply:
Updated the code to use assertIn.